src/functions.php

```diff
@@ -205,28 +207,32 @@ function dissolveGroups(array $vcards): array
     $groups = [];
 
     // separate iCloud groups
+    /** @var \stdClass $vcard */
     foreach ($vcards as $key => $vcard) {
         if (isset($vcard->{'X-ADDRESSBOOKSERVER-KIND'})) {
             if ($vcard->{'X-ADDRESSBOOKSERVER-KIND'} == 'group') {      // identifier
                 foreach ($vcard->{'X-ADDRESSBOOKSERVER-MEMBER'} as $member) {
-                    $groups[(string)$vcard->FN][] = (string)$member;
+                    $member = str_replace(['urn:', 'uuid:'], ['', ''], (string)$member);
+                    $groups[(string)$vcard->FN][] = $member;
                 }
-            unset($vcards[$key]);                                       // delete this vCard
+                unset($vcards[$key]);                                   // delete this vCard
             }
         }
     }
 
-    $vcards = array_values($vcards);
+    $vcards = array_values($vcards);   // no clue, what this line should do
 
     // assign group memberships
     foreach ($vcards as $vcard) {
         foreach ($groups as $group => $members) {
             if (in_array((string)$vcard->UID, $members)) {
-                if (!isset($vcard->GROUP)) {
-                    $vcard->add('GROUP', []);
+                if (isset($vcard->GROUPS)) {
```

Comment:
Groups statt group? Testcase?

Reply:
yes - let´s have a look in FunctionsTest